src/Network/OAuth/OAuth2/Internal.hs

```diff
@@ -63,12 +67,62 @@ instance FromJSON OAuth2Token where
 instance ToJSON OAuth2Token where
     toEncoding = genericToEncoding defaultOptions { fieldLabelModifier = camelTo2 '_' }
 
+-- | Possible OAuth error types
+-- | Not all of them can appear everywhere, but this is good enough for now.
```

Comment:
You could check the spec.https://tools.ietf.org/html/rfc6749
It does indicate Error for each flow.

and here is what I found https://github.com/freizl/hoauth2/blob/hw-rewrite-study/docs/spec-notes.lhs (it's an very rough doc to capture datas in types.)

my understanding would be 
```
AuthError = AuthCodeFlowError | AccessTokenFlowError | UnknownErrorCode ByteString
AuthCodeFlowError = ...
AccessTokenFlowError = ...
```

In additions, beside capture the error code, maybe we shall capture the error_discreption and error_uri as well when they exists.